common/djangoapps/enrollment/views.py

```diff
@@ -783,3 +789,88 @@ def post(self, request):
                     actual_activation=current_enrollment['is_active'] if current_enrollment else None,
                     user_id=user.id
                 )
+
+
+@can_disable_rate_limit
+class EdraakCourseListView(APIView, ApiKeyPermissionMixIn):
+    """
+    """
+    authentication_classes = (JwtAuthentication, OAuth2AuthenticationAllowInactiveUser,
+                              EnrollmentCrossDomainSessionAuth,)
+    permission_classes = ApiKeyHeaderPermissionIsAuthenticated,
+    throttle_classes = EnrollmentUserThrottle,
+    http_method_names = ['get']
+
+    # - Since the course about page on the marketing site uses this API to auto-enroll users,
+    # we need to support cross-domain CSRF.
+    # - (EDRAAK_COURSE_LIST_API_CACHE_SECONDS) is not needed in cms but still being processed
+    # in URL parsing. Therefore, we read it through (getattr) with default of (10 minutes)
+    # - (vary_on_cookie) to create a cache per cookie (thus, per user)
+    @method_decorator(ensure_csrf_cookie_cross_domain)
+    @method_decorator(cache_page(getattr(settings, 'EDRAAK_COURSE_LIST_API_CACHE_SECONDS', 60 * 10)))
```

Comment:
`cache_page` caches the same results for _all_ learners regardless of their permissions.

If it's user-specific and requires authentication, then it's not something good to cache.

Reply:
It is not for (all); below there is `@method_decorator(vary_on_cookie)` which creates a distinct cache for each session. Would you like me to verify that with a test case?

Reply:
Good question Shadi. I don't think we're using a per-user caching. `cache_page` is a [per-site](https://docs.djangoproject.com/en/3.0/topics/cache/#the-per-view-cache) caching and doesn't make sense for an API that aims to provide a user-specific response. Using `vary_on_cookie` is pretty much an anti-pattern in this case and I'll explain why.

If we do per-user caching (e.g. via `vary_on_cookie`) it could overwhelm `memcache` with say 1 million entry which is not a good use of our caching resources since there are better alternatives for speeding up such APIs.

Reply:
Thank you for the explanation @OmarEdraak :+1: